### PR TITLE
Update contributing

### PR DESCRIPTION
**Pull request by naokazuterada**
@jonasbn , hello again!

It's been a long time, but this PR is an measure to deal with the problem we suffered before. https://github.com/naokazuterada/MarkdownTOC/issues/95#issuecomment-304138911

I also added the [issue template](https://blog.github.com/2018-01-25-multiple-issue-and-pull-request-templates/) for common types of issues.

Please take a look whenever you have a time. 

Thanks!

### CODE REVIEW

**Comment by jonasbn** [comment]
Hi @naokazuterada 

I will have a look ASAP.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -4,12 +4,24 @@ These are the guidelines for contributing to this repository.
 
 ## Issues
 
-File an issue if you think you've found a bug. Please describe
+Please follow the each template by the type of your issue.
```
Comment (jonasbn) [inline .github/CONTRIBUTING.md]: "Please follow the each template by the type of you issue." should read: "Please use  the template for your type of issue"

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -4,12 +4,24 @@ These are the guidelines for contributing to this repository.
...
-2. What was expected
-3. What actually occurred
-4. What version of the involved component
+- [Bug Report](https://github.com/naokazuterada/MarkdownTOC/issues/new)
+- [Feature Request](https://github.com/naokazuterada/MarkdownTOC/issues/new?template=feature.md)
+- [Question](https://github.com/naokazuterada/MarkdownTOC/issues/new?template=question.md)
+
+### Way of operation
+
+1. The contributors will `Close` these Issues or PR
```
Comment (jonasbn) [inline .github/CONTRIBUTING.md]: Do you mean maintainers?

Suggestion: "The maintainers will `Close` the issue or PR"

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -4,12 +4,24 @@ These are the guidelines for contributing to this repository.
...
+    1. Not accepted feature request
+        - Add `wontfix` label
```
Comment (jonasbn) [inline .github/CONTRIBUTING.md]: Add the `wontfix`label

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -4,12 +4,24 @@ These are the guidelines for contributing to this repository.
...
+    2. Not obvious and Not responding
```
Comment (jonasbn) [inline .github/CONTRIBUTING.md]: Suggestion: "Not clear or unresponsive"

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -4,12 +4,24 @@ These are the guidelines for contributing to this repository.
...
+        - Not obvious: Problem is not obvious, or not reproduced.
```
Comment (jonasbn) [inline .github/CONTRIBUTING.md]: Suggestion: "Not clear: Problem is not obvious or not reproducible"

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -4,12 +4,24 @@ These are the guidelines for contributing to this repository.
...
+        - Not responding: Author doesn't respond in 3 weeks.
```
Comment (jonasbn) [inline .github/CONTRIBUTING.md]: Suggestion: "Unresponsive: the author does not respond to inquiries within 3 weeks."

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -4,12 +4,24 @@ These are the guidelines for contributing to this repository.
...
+
+## Gitter
+
+If you want to ask more easily, please use [Gitter](https://gitter.im/naokazuterada/MarkdownTOC) chat system.
```
Comment (jonasbn) [inline .github/CONTRIBUTING.md]: Suggestion: "If you want more general support or to ask questions, please use ..."

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -21,11 +33,15 @@ You should use unit-tests for SublimeText by using [UnitTesting](https://github.
...
 2. Comment out or rename your own `MarkdownTOC.sublime-settings` so individual settings are not used during testing
-3. [Run tests](https://github.com/randy3k/UnitTesting-example#running-tests) 
+3. [Run tests](https://github.com/randy3k/UnitTesting-example#running-tests)
 4. Send Pull Request when tests pass
 
 All contributions are welcome
 
+## Useful Tools
+
+- [Gyazo](https://gyazo.com/en): Tool for making screen captures into animation GIF. Use it for showing how the function works.
```
Comment (jonasbn) [inline .github/CONTRIBUTING.md]: Suggestion: "Tool for making animated GIFs from screen captures".

File: .github/ISSUE_TEMPLATE.md [hunk 9]
```diff
@@ -1,8 +1,25 @@
...
-4. What version of the involved component
--->
\ No newline at end of file
+Template: BUG REPORT
+
+This is an issue for a bug report.
+You should label `bug-report`.
+
+
+If your purpose is different, please visit following links.
```
Comment (jonasbn) [inline .github/ISSUE_TEMPLATE.md]: Suggestion: "If you issue is not a bug report, please visit one of the following links."

File: .github/ISSUE_TEMPLATE.md [hunk 10]
```diff
@@ -1,8 +1,25 @@
...
 
-1. How can it be reproduced
-2. What was expected
-3. What actually occurred
-4. What version of the involved component
--->
\ No newline at end of file
+Template: BUG REPORT
+
+This is an issue for a bug report.
```
Comment (jonasbn) [inline .github/ISSUE_TEMPLATE.md]: Suggestion: Replace "issue" with "template"

File: .github/ISSUE_TEMPLATE.md [hunk 11]
```diff
@@ -1,8 +1,25 @@
...
+
+FEATURE REQUEST:
+https://github.com/naokazuterada/MarkdownTOC/issues/new?template=feature.md
+
+QUESTION:
+https://github.com/naokazuterada/MarkdownTOC/issues/new?template=question.md
+
+-->
+
+# How can it be reproduced
```
Comment (jonasbn) [inline .github/ISSUE_TEMPLATE.md]: Suggestion: "How can the issue be reproduced"

File: .github/ISSUE_TEMPLATE.md [hunk 12]
```diff
@@ -1,8 +1,25 @@
...
+
+# What was expected
+
+# What actually occurred
+
+# What version of the involved component
```
Comment (jonasbn) [inline .github/ISSUE_TEMPLATE.md]: Suggestion: "What was the version of the involved component"

File: .github/ISSUE_TEMPLATE/feature.md [hunk 13]
```diff
@@ -0,0 +1,23 @@
+<!--
+
+Template: FEATURE REQUEST
+
+This is an issue for a feature request.
+You should label `feature-request`.
+
+
+If your purpose is different, please visit following links.
```
Comment (jonasbn) [inline .github/ISSUE_TEMPLATE/feature.md]: Suggestion: "If you issue is not a feature request, please visit one of the following links."

File: .github/ISSUE_TEMPLATE/question.md [hunk 14]
```diff
@@ -0,0 +1,19 @@
+<!--
+
+Template: QUESTION
+
+This is an issue for a feature request.
```
Comment (jonasbn) [inline .github/ISSUE_TEMPLATE/question.md]: Suggestion: "This is a template for a general question"

File: .github/ISSUE_TEMPLATE/feature.md [hunk 15]
```diff
@@ -0,0 +1,23 @@
+<!--
+
+Template: FEATURE REQUEST
+
+This is an issue for a feature request.
```
Comment (jonasbn) [inline .github/ISSUE_TEMPLATE/feature.md]: Suggestion: Replace "issue" with "template"

File: .github/ISSUE_TEMPLATE/question.md [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+You should label `question`.
+
+
+If your purpose is different, please visit following links.
```
Comment (jonasbn) [inline .github/ISSUE_TEMPLATE/question.md]: Suggestion: "If you issue is not a question, please visit one of the following links."

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (naokazuterada) [reply .github/CONTRIBUTING.md]: 👍

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (naokazuterada) [reply .github/CONTRIBUTING.md]: Thanks you are right.

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (naokazuterada) [reply .github/CONTRIBUTING.md]: 👍

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (naokazuterada) [reply .github/CONTRIBUTING.md]: 👍

File: .github/CONTRIBUTING.md (hunk 5 above)
Comment (naokazuterada) [reply .github/CONTRIBUTING.md]: 👍

File: .github/CONTRIBUTING.md (hunk 6 above)
Comment (naokazuterada) [reply .github/CONTRIBUTING.md]: great! 🌟

File: .github/CONTRIBUTING.md (hunk 7 above)
Comment (naokazuterada) [reply .github/CONTRIBUTING.md]: 👍

File: .github/CONTRIBUTING.md (hunk 8 above)
Comment (naokazuterada) [reply .github/CONTRIBUTING.md]: thanks!

File: .github/ISSUE_TEMPLATE.md (hunk 9 above)
Comment (naokazuterada) [reply .github/ISSUE_TEMPLATE.md]: Did you mean "If your issue is..."? 🙄

File: .github/ISSUE_TEMPLATE.md (hunk 10 above)
Comment (naokazuterada) [reply .github/ISSUE_TEMPLATE.md]: 👍

File: .github/ISSUE_TEMPLATE.md (hunk 11 above)
Comment (naokazuterada) [reply .github/ISSUE_TEMPLATE.md]: thanks! :sweat_drops:

File: .github/ISSUE_TEMPLATE.md (hunk 12 above)
Comment (naokazuterada) [reply .github/ISSUE_TEMPLATE.md]: 👍

File: .github/ISSUE_TEMPLATE/feature.md (hunk 13 above)
Comment (naokazuterada) [reply .github/ISSUE_TEMPLATE/feature.md]: 👍

File: .github/ISSUE_TEMPLATE/question.md (hunk 14 above)
Comment (naokazuterada) [reply .github/ISSUE_TEMPLATE/question.md]: Oh... sorry, 😓

File: .github/ISSUE_TEMPLATE/feature.md (hunk 15 above)
Comment (naokazuterada) [reply .github/ISSUE_TEMPLATE/feature.md]: great!

File: .github/ISSUE_TEMPLATE/question.md (hunk 16 above)
Comment (naokazuterada) [reply .github/ISSUE_TEMPLATE/question.md]: 👍

**Comment by naokazuterada** [comment]
@jonasbn Thank you so much!! 

I correct almost everything you pointed out. 

I thought [this](https://github.com/naokazuterada/MarkdownTOC/pull/121#discussion_r175562437) might be a miss type.

I'm so appreciate for compensating for my lack of English skills! Very helpful!

File: .github/ISSUE_TEMPLATE.md (hunk 9 above)
Comment (jonasbn) [reply .github/ISSUE_TEMPLATE.md]: yes of course - I think I was getting tired at that point 👍

Comment (naokazuterada) [reply .github/ISSUE_TEMPLATE.md]: 👍 👍 👍

**Pull request merged**